@types/react-dom 15.5.2 causes typescript build break by adding @types/node dependency

Today my builds started failing because of a conflict between requirejs and node require

This is the compile error I received from typescript:
```
d:/a/1/s/node_modules/@types/node/index.d.ts(99,13): 
error TS2403: Subsequent variable declarations must have the same type.  
Variable 'require' must be of type 'Require', but here has type 'NodeRequire'.
```
Lookin at the npm install diff since last known good built showed that it was introduced from react-dom:
![image](https://user-images.githubusercontent.com/9914123/28945102-eb720cf8-7859-11e7-8858-4409e1aa336e.png)

15.5.1 didn't have the dependency to `@types/node`

```bash
node_modules/@types/node/index.d.ts(73,13): error TS2451: Cannot redeclare block-scoped variable 'global'.
node_modules/@types/node/index.d.ts(99,13): error TS2300: Duplicate identifier 'require'.
node_modules/@types/react-native/index.d.ts(9036,11): error TS2451: Cannot redeclare block-scoped variable 'global'.
node_modules/@types/react-native/index.d.ts(9037,14): error TS2300: Duplicate identifier 'require'.
```

--- a/types/react-dom/node-stream/index.d.ts
+++ b/types/react-dom/node-stream/index.d.ts
@@ -1,20 +1,18 @@
-/// <reference types="node" />
 import { ReactElement } from 'react';
-import * as stream from 'stream';
 
 /**
  * Render a ReactElement to its initial HTML. This should only be used on the
  * server.
  * See https://facebook.github.io/react/docs/react-dom-stream.html#rendertostream
  */
-export function renderToStream(element: ReactElement<any>): stream.Readable;
+export function renderToStream(element: ReactElement<any>): any;
 
 /**
  * Similar to renderToStream, except this doesn't create extra DOM attributes
  * such as data-react-id that React uses internally.
  * See https://facebook.github.io/react/docs/react-dom-stream.html#rendertostaticstream
  */
-export function renderToStaticStream(element: ReactElement<any>): stream.Readable;
+export function renderToStaticStream(element: ReactElement<any>): any;
 export const version: string;
 
 export as namespace ReactDOMNodeStream;

--- a/types/react-dom/react-dom-tests.ts
+++ b/types/react-dom/react-dom-tests.ts
@@ -3,7 +3,6 @@ import * as ReactDOM from 'react-dom';
 import * as ReactDOMServer from 'react-dom/server';
 import * as ReactDOMNodeStream from 'react-dom/node-stream';
 import * as ReactTestUtils from 'react-dom/test-utils';
-import * as stream from 'stream';
 
 declare function describe(desc: string, f: () => void): void;
 declare function it(desc: string, f: () => void): void;
@@ -41,11 +40,11 @@ describe('ReactDOMServer', () => {
 
 describe('ReactDOMNodeStream', () => {
     it('renderToStream', () => {
-        const content: stream.Readable = ReactDOMNodeStream.renderToStream(React.createElement('div'));
+        const content: any = ReactDOMNodeStream.renderToStream(React.createElement('div'));
     });
 
     it('renderToStaticStream', () => {
-        const content: stream.Readable = ReactDOMNodeStream.renderToStaticStream(React.createElement('div'));
+        const content: any = ReactDOMNodeStream.renderToStaticStream(React.createElement('div'));
     });
 });
 

